src/ucp/tag/eager_snd.c

```diff
@@ -203,7 +203,18 @@ void ucp_tag_eager_zcopy_completion(uct_completion_t *self,
                                     ucs_status_t status)
 {
     ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct_comp);
-    ucp_tag_eager_zcopy_req_complete(req, status);
+    if (req->send.state.offset == req->send.length) {
+        ucp_tag_eager_zcopy_req_complete(req, status);
+    } else if (status != UCS_OK) {
+        /* NOTE: the request is in pending queue if data was not completely sent,
+         *       just dereg the buffer here and complete request on purge
+         *       pending later. The same for RNDV and SYNC protocols.
+         * TODO: add error completion in ucp_proto_t to handle failure of
```

Comment:
let's unite code with ucp_tag_eager_sync_zcopy_req_complete - e.g add function like ucp_tag_zcopy_failure()
used in:
1. here
2. ucp_tag_eager_sync_zcopy_req_complete
3. ucp_rndv_contig_zcopy_completion